lib/src/Utilities.cc

```diff
@@ -395,7 +395,22 @@ std::string getUuid()
 #else
     uuid_t uu;
     uuid_generate(uu);
-    return binaryStringToHex(uu, 16);
+    std::string hexUuid{binaryStringToHex(uu, 16)};
+
```

Comment:
I prefer to change `std::string binaryStringToHex(const unsigned char *ptr, size_t length)` to `std::string binaryStringToHex(const unsigned char *ptr, size_t length, bool lowerCase=false)`

Reply:
ok, but what about the token hyphens, continue in the std::string getUuid() function?

Reply:
How about:
```c++
uuid.append(binaryStringToHex(uu,8,true)).append('-').append(uu+8,4,true)....
```